libraries/Parsers/Microsoft.Bot.Builder.Parsers.LU/Parser/LUResource.cs

```diff
@@ -0,0 +1,407 @@
+
+        [JsonProperty("Roles", NullValueHandling = NullValueHandling.Ignore)]
+        public string Roles { get; set; }
+
+        [JsonProperty("ListBody", NullValueHandling = NullValueHandling.Ignore)]
+        public List<string> ListBody { get; set; }
+
+        [JsonProperty("CompositeDefinition", NullValueHandling = NullValueHandling.Ignore)]
+        public string CompositeDefinition { get; set; }
+
+        [JsonProperty("RegexDefinition", NullValueHandling = NullValueHandling.Ignore)]
+        public string RegexDefinition { get; set; }
+
+        [JsonProperty("SynonymsOrPhraseList", NullValueHandling = NullValueHandling.Ignore)]
+        public List<string> SynonymsOrPhraseList { get; set; }
+
+        [JsonProperty("Features", NullValueHandling = NullValueHandling.Ignore)]
+        public string Features { get; set; }
+
+        [JsonProperty("SynonymsList", NullValueHandling = NullValueHandling.Ignore)]
+        public List<NewEntitySection.SynonymElement> SynonymsList { get; set; } = null;
+    }
+
+    public partial class UtteranceAndEntitiesMap
+    {
+        [JsonProperty("utterance")]
+        public string Utterance { get; set; }
+
+        [JsonProperty("entities")]
+        public List<EntityElement> Entities { get; set; }
+
+        [JsonProperty("errorMsgs")]
+        public List<string> ErrorMsgs { get; set; }
+
+        [JsonProperty("contextText")]
+        public string ContextText { get; set; }
+
+        [JsonProperty("range")]
+        public Range Range { get; set; }
+
+        [JsonProperty("references", NullValueHandling = NullValueHandling.Ignore)]
+        public Reference References { get; set; }
+    }
+
+    public partial class Reference
+    {
+        [JsonProperty("source")]
+        public string Source { get; set; }
+    }
+
+    public partial class EntityElement
+    {
+        [JsonProperty("type")]
+        [JsonConverter(typeof(StringEnumConverter))]
+        public TypeEnum Type { get; set; }
+
+        [JsonProperty("entity")]
+        public string Entity { get; set; }
+
+        [JsonProperty("role")]
+        public string Role { get; set; }
+
+        [JsonProperty("startPos", NullValueHandling = NullValueHandling.Ignore)]
+        public int? StartPos { get; set; }
+
+        [JsonProperty("endPos", NullValueHandling = NullValueHandling.Ignore)]
+        public int? EndPos { get; set; }
+    }
+
+    internal static class Converter
+    {
+        public static readonly JsonSerializerSettings Settings = new JsonSerializerSettings
+        {
+            MetadataPropertyHandling = MetadataPropertyHandling.Ignore,
+            DateParseHandling = DateParseHandling.None,
+            Converters =
+            {
+                SectionTypeConverter.Singleton,
+                TypeEnumConverter.Singleton,
+                new IsoDateTimeConverter { DateTimeStyles = DateTimeStyles.AssumeUniversal }
+            },
+        };
+    }
+
+    internal class SectionTypeConverter : JsonConverter
```

Comment:
I don't think we should have this in the same file as the other stuff. One class per file unless auto-generated please.

Reply:
Also comments  / docs aboutr why we need a custom converter as opposed to the built in json converter would be handy

Reply:
This was auto-generated